edx_when/api.py

```diff
@@ -398,11 +398,29 @@ def get_schedules_with_due_date(course_id, assignment_date):
     # Using those relative dates, get all Schedules that have a "hit" by working backwards to the start_date
     rel_start_dates = [assignment_date - rel_date for rel_date in rel_dates]
 
-    # Exclude any user that has an overridden date for a course on the specified day so there aren't duplicates
-    return Schedule.objects.filter(
-        enrollment__course_id=course_id,
-        start_date__date__in=rel_start_dates,
-    ).exclude(enrollment__user_id__in=user_ids).select_related('enrollment') | user_date_overridden_schedules
+    if rel_start_dates:
```

Comment:
You don't *need* this if do you? Wouldn't the ORM or MySQL know to not do anything for a `where x in ()` clause? Maybe doesn't matter here, where the check is cheep. But below it's an extra query on ContentDate.

Reply:
Naw the `if` isn't necessary, but if I can avoid querying the DB I might as well since I know I shouldn't expect results if that condition is met